### PR TITLE
Fix String.rsplit to return splitted list with correct order

### LINKED ISSUE #36898: string.rsplit not returning array from right delimiter
**Godot version:**
3.2

**OS/device including version:**
Windows 10

**Issue description:**

https://docs.godotengine.org/en/3.2/classes/class_string.html#class-string-method-rsplit

string.rsplit should [quote] "returns a array of the substrings, starting from right."
So [0] element of the array should be the right hand most substring
Instead it returns from the left

**Steps to reproduce:**
Copy the example code from the documentation:

var some_string = "One,Two,Three,Four"
var some_array = some_string.rsplit(",", true, 1)
print(some_array.size()) # Prints 2
print(some_array[0]) # Prints "Four"                     #### actually prints "One,Two,Three"
print(some_array[1]) # Prints "Three,Two,One"    #### actually prints "Four"

**Minimal reproduction project:**

### PR DESCRIPTION
**Pull request by dreamsComeTrue**
With PR below code (copied from documentation) finally works as expected:

https://docs.godotengine.org/en/3.2/classes/class_string.html#class-string-method-rsplit

```gdscript
var some_string = "One,Two,Three,Four"
var some_array = some_string.rsplit(",", true, 1)
print(some_array.size()) # Prints 2
print(some_array[0]) # Prints "Four"
print(some_array[1]) # Prints "Three,Two,One"
```

Fixes: #36898

### CODE REVIEW

File: core/ustring.cpp [hunk 1]
```diff
@@ -838,19 +840,37 @@ Vector<String> String::rsplit(const String &p_splitter, bool p_allow_empty, int
...
-	ret.invert();
+	if (!sub_part_left.empty()) {
+		if (p_maxsplit > 0 && p_maxsplit <= ret.size()) {
+			split_str_accum += sub_part_left;
+		} else {
+			ret.push_back(sub_part_left);
+		}
+	}
+
+	//	If 'p_maxsplit' is used, add remaining items at tail as one long String - reversed and delimiter-separated
```
Comment (akien-mga) [inline core/ustring.cpp]: Comments should start with a space, not a tab.

Comment (dreamsComeTrue) [reply core/ustring.cpp]: Thanks for noticing that - I thought my auto-format capabilities in VSCode already handle that as well - but will remember for the future ;)

**Comment by dankan1890** [comment]
I think the output of the example is wrong, the correct output should be:
```
print (some_array [0]) # Prints "Four"
print (some_array [1]) # Prints "One, Two, Three"
```
or am I wrong?

**Comment by Anutrix** [comment]
> 
> 
> I think the output of the example is wrong, the correct output should be:
> 
> ```
> print (some_array [0]) # Prints "Four"
> print (some_array [1]) # Prints "One, Two, Three"
> ```
> 
> or am I wrong?

I agree because:
> The splits in the returned array are sorted in the same order as the original string, from left to right.

**Comment by kleonc** [comment]
> With PR below code (copied from documentation) finally works as expected:
> https://docs.godotengine.org/en/3.2/classes/class_string.html#class-string-method-rsplit
> ```gdscript
> var some_string = "One,Two,Three,Four"
> var some_array = some_string.rsplit(",", true, 1)
> print(some_array.size()) # Prints 2
> print(some_array[0]) # Prints "Four"
> print(some_array[1]) # Prints "Three,Two,One"
> ```

It shouldn't work like that. This PR shouldn't be merged as is.

It should work either as @dankan1890 suggested (without the additional spaces):
```
print (some_array[0]) # Prints "Four"
print (some_array[1]) # Prints "One,Two,Three"
```
or as @Anutrix suggested (without the additional spaces), that's how it works currently (the docs are just wrong):
```
print(some_array[0]) # Prints "One,Two,Three"
print(some_array[1]) # Prints "Four"
```

**Comment by akien-mga** [comment]
Superseded by #59547.

**Pull request closed without merging**